src/hardhat.ts

```diff
@@ -0,0 +1,68 @@
+import { CHAINS } from './generated/chains';
+import { toUpperSnakeCase } from './utils/strings';
+import { Chain, HardhatEtherscanConfig, HardhatNetworksConfig } from './types';
+
+export function buildEnvVariables(): string[] {
+  return CHAINS
+    .filter((chain) => chain.explorer?.api?.key?.required)
+    .map((chain) => buildEtherscanApiKeyName(chain));
+}
+
+export function buildEtherscanApiKeyName(chain: Chain): string {
+  return `${toUpperSnakeCase(chain.alias)}_ETHERSCAN_API_KEY`;
+}
+
+// https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify#multiple-api-keys-and-alternative-block-explorers
+export function buildEtherscanConfig(): HardhatEtherscanConfig {
+  // Not usable outside of a Node.js environment
+  if (typeof window !== 'undefined') {
+    return { apiKey: {}, customChains: [] };
+  }
+
+  return CHAINS.reduce((etherscan, chain) => {
+    if (!chain.explorer || !chain.explorer.api) {
+      return etherscan;
+    }
+
+    const explorer = chain.explorer;
+    const apiKey = chain.explorer.api.key;
+
+    const apiKeyEnvName = buildEtherscanApiKeyName(chain);
+    const apiKeyValue = apiKey.required ? chain.alias : process.env[apiKeyEnvName];
+
+    if (apiKey.hardhatEtherscanAlias) {
+      etherscan.apiKey[apiKey.hardhatEtherscanAlias] = apiKeyValue || '';
+      return etherscan;
+    }
+
+    etherscan.customChains.push({
+      network: chain.alias,
+      chainId: Number(chain.id),
+      urls: {
+        apiURL: explorer.api?.url || '',
+        browserURL: explorer.browserUrl,
+      },
+    });
+
+    etherscan.apiKey[chain.alias] = apiKeyValue || '';
```

Comment:
Isn't `apiKeyValue` already defined above? I don't get the `|| ''`.

Reply:
`process.env['...']` returns `string | undefined`. I've changed the `apiKeyValue` to be

```ts
const apiKeyValue = apiKey.required ? (process.env[apiKeyEnvName] || 'NOT_FOUND') : 'DUMMY_VALUE';
```
so the `|| ''` can now be dropped (let me know if you prefer `DUMMY_VALUE` in both places)

Reply:
I like `const apiKeyValue = apiKey.required ? (process.env[apiKeyEnvName] || 'NOT_FOUND') : 'DUMMY_VALUE';`

I don't understand why you still haven't dropped the `|| ''` though

Reply:
Thanks, it was in two places and I only updated the one. Fixed here: https://github.com/api3dao/chains/pull/44/commits/1875c954abbf82ac3d53074f44b8074d209ad494